client/src/telethon/_impl/mtsender/sender.py

```diff
@@ -192,62 +187,61 @@ async def connect(
         return cls(
             dc_id=dc_id,
             addr=addr,
-            lock=Lock(),
             _logger=base_logger.getChild("mtsender"),
+            _lock=Lock(),
             _reader=reader,
             _writer=writer,
             _transport=transport,
             _mtp=mtp,
-            _mtp_buffer=bytearray(),
-            _requests=[],
-            _request_event=Event(),
+            _requests=Queue(1),
```

Comment:
I'm not sure why there's a limit of a single request. This does enable backpressure, but it might make it harder for the library to be able to combine multiple requests into a single container.